dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java

```diff
@@ -223,6 +225,12 @@ private void addCollectionEntriesToCollection(Collection collection) {
         collectionEntries.addAll(collectionWorkflowsWithVersions);
         collectionEntries.addAll(collectionServicesWithVersions);
         collectionEntries.addAll(collectionToolsWithVersions);
+        collectionEntries.forEach(entry -> {
+            //Object entryDAO;
+            List<Label> labels = workflowDAO.getLabelByEntryId(entry.getId());
```

Comment:
This would be more efficient if you made it `workflowDAO.getLabelsByEntryIds(<entry ids>)`. An intermediate step towards annotations, perhaps.

Reply:
Ah I see, but in this way, I will still need to parse all the labels later on, since I wish to have labels stored in their respective entry

Reply:
> Ah I see, but in this way, I will still need to parse all the labels later on, since I wish to have labels stored in their respective entry

Yes, you would. But if a collection has 10 workflows, you would execute 10 SQL statements, vs. 1 statement with a little extra work to do some Java parsing. But if this isn't called when you fetch organizations, which I don't think it is, then it's OK.

Reply:
This won't easily work either. Best to ignore this comment also. I had been thinking in SQL terms, and had forgotten some of the JPA limitations. :)